pandas/core/tools/datetimes.py

```diff
@@ -212,7 +212,7 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
         if box:
             if errors == 'ignore':
                 from pandas import Index
-                return Index(result, name=name)
+                return Index(result, name=name, tz=tz)
```

Comment:
While I think this works, we want to avoid passing this directly in the constructor.

Instead, if this line results in a DatetimeIndex, we want to use `tz_localize(tz)` instead.